src/pymoca/tree.py

```diff
@@ -679,11 +679,74 @@ def _find_imported(
     check_builtin_classes: bool = False,
     current_extends: Optional[Set[ast.ExtendsClause]] = None,
 ) -> Optional[Union[ast.Class, ast.Symbol]]:
-    """Find simple name in imports"""
-    # TODO: Imported name lookup
+    """Find simple name in imports per MLS v3.5 section 13.2.1"""
+    # TODO: Rewrite this to work with parser rewrite of import_clause handler.
+    # TODO: Can we do a scope.imports[name] = found Class or Symbol to speed up future calls?
+    # Search qualified imports (most common case)
+    if name in scope.imports:
+        import_: Union[ast.ImportClause, ast.ComponentRef] = scope.imports[name]
+        if isinstance(import_, ast.ImportClause):
+            import_ = import_.components[0]
+        found = find_name(
+            import_,
+            scope.root,
+            check_builtin_classes=check_builtin_classes,
+            search_parent=False,
+            copy=False,
+        )
+        _check_import_rules(found, scope)
+        return found
+    # Unqualified imports
+    if "*" in scope.imports:
+        c = None
+        for package_ref in scope.imports["*"].components:
+            imported_comp_ref = package_ref.concatenate(ast.ComponentRef(name=name))
+            # Search within the package
+            # Avoid infinite recursion with search_imports = False
+            c = find_name(
+                imported_comp_ref,
+                scope.root,
+                search_imports=False,
+                search_parent=False,
+                check_builtin_classes=check_builtin_classes,
+                current_extends=current_extends,
+                copy=False,
+            )
+            _check_import_rules(c, scope)
+            if c is not None:
+                # Store result for next lookup
+                scope.imports[name] = imported_comp_ref
+                return c
     return None
 
 
+def _check_import_rules(
+    element: Optional[Union[ast.Class, ast.Symbol]],
+    scope: ast.Class,
+) -> None:
+    """Check import rules per the Modelica spec"""
+    if element is None:
+        return
+    if not element.parent:
+        raise NameLookupError(f"Import {element.name} must be contained in a package")
+    if element.parent.type != "package":
```

Comment:
I think (but have not tested) that you can define a `package` inside a `class`. The spec section 13.2.1.1 on name lookup only says check the immediate parent:
> Qualified import clauses may only refer to packages or elements of packages, i.e., in import A.B.C; or
import D=A.B.C;, A.B must be a package. Unqualified import clauses may only import from packages, i.e., in
import A.B.*;, A.B must be a package. [Note: in import A; the class A can be any class which is an element
of the unnamed top-level package]

Reply:
I agree these checks need better testing. Here is a first pass at a test case:

```Modelica
class C
  package P
    constant Real a;
  end P;
  model M
    constant Real a;
  end M;
  model ShouldPass
    import P.a;
    Real b = P.a;
  end ShouldPass;
  model ShouldFail
    import M.a;
    Real b = M.a;
  end ShouldFail;
end C;
```

OMC flattens both `ShouldPass` and `ShouldFail` without errors. However, I think `ShouldFail` should fail :smile: because we are importing element `a` from a `model`, not a `package`. Also note that `P` is contained in a non-package, which I think is OK, but it appears that we can't really depend on OMC error checking for verification.
```suggestion
    if element.parent.type != "package":
```

Also TODO: The `if not element.parent: raise NameLookupError(f"Import {element.name} must be contained in a package")` check looks suspicious. I think `not element.parent` is *not* a sufficient check of element.parent.
Let's try to get all patch lines covered before we merge with the main aim of preventing bugs (but not perfect Modelica language error checking for now).

Reply:
We already have test cases similar to the above from `ModelicaCompliance` in test_name_lookup.py.

We do not have test that packages can only have package parents because it is not a thing as far as I can tell.

I think this is answered well enough so going to call it resolved.